src/dolphin/atmosphere/troposphere.py

```diff
@@ -0,0 +1,557 @@
+from __future__ import annotations
+
+import datetime
+import os
+from dataclasses import dataclass
+from os import fspath
+from pathlib import Path
+from typing import Literal, Optional, Union
+
+import numpy as np
+import opera_utils as oput
+import pyaps3 as pa
+from osgeo import gdal
+from pyproj import CRS
+from scipy.interpolate import RegularGridInterpolator
+
+from dolphin import io, stitching
+from dolphin._dates import _format_date_pair, get_dates
+from dolphin._log import get_log
+from dolphin._types import Bbox, Filename
+
+logger = get_log(__name__)
+
+
+###########
+
+
```

Comment:
perhaps we can add some Enums to make the typing easier, and to better advertise which choices we support:

```python
from enum import StrEnum


class WeatherModel(StrEnum):
    """Type of weather model available for PyAPS/RAIDER."""

    ERA5 = "ERA5"
    ERAI = "ERAI"
    MERRA2 = "MERRA2"
    NARR = "NARR"
    HRRR = "HRRR"
    GMAO = "GMAO"
    HRES = "HRES"


class DelayType(StrEnum):
    """Type of tropospheric delay to calculate."""

    WET = "wet"
    DRY = "dry"
    COMB = "comb"
```

The nice part about the `StrEnum` is that the we dont have to change much to work with the downstream libraries that might be expecting a `str` with a certain value.

```python
In [9]: WeatherModel.ERA5
Out[9]: <WeatherModel.ERA5: 'ERA5'>

In [10]: isinstance(WeatherModel.ERA5, str)
Out[10]: True

In [11]: d = {'ERA5': 123}

# we can access `dict`s that were made with the string as key
In [12]: d[WeatherModel.ERA5]
Out[12]: 123

# If we need all the strings:

In [16]: [w.value for w in WeatherModel]
Out[16]: ['ERA5', 'ERAI', 'MERRA2', 'NARR', 'HRRR', 'GMAO', 'HRES']
```

(note that i'm naming the member variables all caps due to this advice: https://mail.python.org/pipermail/python-ideas/2016-September/042340.html)

Reply:
Thanks @scottstanie, so we are not doing this because of python version as we discussed offline